docs/user/adding-glean-to-your-project.md

```diff
@@ -266,6 +266,10 @@ Please refer to the [custom pings documentation](pings/custom.md).
 > **Important**: as stated [before](adding-glean-to-your-project.md#before-using-glean), any new data collection requires documentation and data-review.
 > This is also required for any new metric automatically collected by the Glean SDK.
 
+### Parallelism
+
+All of Glean's target languages use a separate worker thread to do most of Glean's work, including any I/O. This thread is fully managed by Glean as an implementation detail. Therefore, users should be free to use the Glean API wherever it is most convenient, without worrying about the performance impact of updating metrics and sending pings.
```

Comment:
I wonder if that's the best place to put it. I have no better suggestion though, as it looks kind of off even in the "The General API" page.

Reply:
Yeah, I struggled with where to put it myself.  It's sort of a "general design principles" thing, but we don't have that, and I don't think there's much that's user-relevant besides this...

Reply:
Yes, I agree :-\ Let's leave it where it is for now, and revisit in the future, in case we add something else!